<!-- This launchfile runs our simple imu+wheel odometry for the warthog. It requires odometry messages from the wheels and the processed IMU attitude
-->
<launch>
  <set_parameter name="use_sim_time" value="true"/>
  <node pkg="norlab_imu_tools" exec="imu_and_wheel_odom_node" name="imu_and_wheel_odom_node" output="screen" respawn="true">
    <remap from="imu_topic" to="/MTI_imu/data" />
    <remap from="wheel_odom_topic" to="/warthog_velocity_controller/odom" />
    <remap from="altitude_topic" to="/warthog_dps310_1_reference_dps310_sphere/altitude" />
    <param name="odom_frame" value="odom" />
    <param name="base_frame" value="base_link" />
    <param name="altimeter_frame" value="dps310_1" />
    <param name="publish_translation" value="True" />
    <param name="wheel_odom_velocity_scale_x" value="0.6332" /> <!-- Value for wheels : 0.95, Value for tracks: 0.6332 -->

    ## Values in RAD
    <param name="mag_north_correction_yaw" value="0.0" />


    ## If odom message needed
    <param name="publish_odom" value="True" />
    <param name="odom_topic_name" value="imu_and_wheel_odom" />

  </node>

</launch>
